ansible_base/lib/abstract_models/common.py

```diff
@@ -66,27 +81,46 @@ def _attributable_user(self, warn_nonexistent_system_user):
                     user = get_user_model().objects.get(username=system_username)
                 except get_user_model().DoesNotExist:
                     if warn_nonexistent_system_user:
-                        logger.warn(f"SYSTEM_USERNAME is set to {system_username} but no user with that username exists. User attribution will be None.")
+                        logger.warn(f"SYSTEM_USERNAME is set to {system_username} but no user with that username exists.")
                     user = None
+
         return user
 
+    def _check_user(self, user, warn_nonexistent_system_user=True):
+        # Its possible for _attributable_user to return None or anonymous user in which case we don't want to save
+        if (not user or user.is_anonymous) and warn_nonexistent_system_user:
+            # TODO: See if there is a better way to figure out how to identify this object?
+            # Maybe instead of trying to send a single identifier we try and dump the object?
+            obj_id = getattr(self, 'pk', getattr(self, 'name', getattr(self, 'username', 'Unknown')))
+            logger.error(f"Request to save a {self.__class__} object, id {obj_id} without a user making the request!")
+            raise ValueError("Unable to save model without user!")
+
     def save(self, *args, warn_nonexistent_system_user=True, **kwargs):
         update_fields = list(kwargs.get('update_fields', []))
-        user = self._attributable_user(warn_nonexistent_system_user)
 
         # Manually perform auto_now_add and auto_now logic.
         now = timezone.now()
-        if not self.pk and not self.created_on:
-            self.created_on = now
+
+        user = self._attributable_user(warn_nonexistent_system_user)
+
+        if self.created_by is None:
+            self._check_user(user, warn_nonexistent_system_user)
```

Comment:
what is the change related to state before? The variable "warn" doesn't cause warning (which is quite confusing), but raises exception. 
And is True by default - is there some `except ValueError` I don't see which makes the difference?

Reply:
Good call on the name. Before it just warned but now it actually raises so I'll update the name.

Reply:
thanks for explanation, it's clear to me now